src/Api/FilterInterface.php

```diff
@@ -36,4 +36,13 @@
      * @return array
      */
     public function getDescription(string $resourceClass): array;
+
+    /**
+     * setter for properties.
+     *
+     * used in the dependency injection to allow independent constructor arguments.
+     *
+     * @param array $properties
+     */
+    public function setProperties(array $properties);
```

Comment:
Fixed this by adding a new Interface (should be merged with Api\FilterInterface if we can break this later).